src/hooks/useSwapInputHandlers.ts

```diff
@@ -28,16 +31,46 @@ export default function useSwapInputHandlers() {
     if (type === ExchangeModalTypes.withdrawal) {
       dispatch(updateSwapInputAmount(supplyBalanceUnderlying));
     } else {
-      let amount =
-        ethereumUtils.getAccountAsset(inputCurrencyAddress)?.balance?.amount ??
-        '0';
-      if (inputCurrencyAddress === ETH_ADDRESS) {
-        const remaining = subtract(amount, MIN_ETH);
-        amount = greaterThan(remaining, 0) ? remaining : '0';
+      const accountAsset = ethereumUtils.getAccountAsset(inputCurrencyAddress);
+      const oldAmount = accountAsset?.balance?.amount ?? '0';
+      let newAmount = oldAmount;
+      if (inputCurrencyAddress === ETH_ADDRESS && accountAsset) {
+        newAmount = toFixedDecimals(
+          ethereumUtils.getBalanceAmount(selectedGasFee, accountAsset),
+          6
+        );
+
+        Alert({
+          buttons: [
+            { style: 'cancel', text: 'No thanks' },
+            {
+              onPress: () => {
+                const transactionFee =
+                  parseFloat(oldAmount) - parseFloat(newAmount);
```

Comment:
can we use the `subtract` function from `helpers/utilities` for these types of operations in general (over parseFloat or Number etc)